core/src/main/java/org/elasticsearch/index/mapper/fixed/FixedPointFieldMapper.java

```diff
@@ -0,0 +1,399 @@
+import org.apache.lucene.util.BytesRefBuilder;
+import org.apache.lucene.util.NumericUtils;
+import org.elasticsearch.Version;
+import org.elasticsearch.action.fieldstats.FieldStats;
+import org.elasticsearch.common.Explicit;
+import org.elasticsearch.common.Numbers;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.Fuzziness;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.index.analysis.NamedAnalyzer;
+import org.elasticsearch.index.analysis.NumericLongAnalyzer;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.Mapper;
+import org.elasticsearch.index.mapper.MapperParsingException;
+import org.elasticsearch.index.mapper.ParseContext;
+import org.elasticsearch.index.mapper.core.NumberFieldMapper;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.common.xcontent.support.XContentMapValues.nodeLongValue;
+import static org.elasticsearch.index.mapper.MapperBuilders.fixedPointField;
+import static org.elasticsearch.index.mapper.MapperBuilders.longField;
+import static org.elasticsearch.index.mapper.core.TypeParsers.parseNumberField;
+
+/**
+ *
+ */
+public class FixedPointFieldMapper extends NumberFieldMapper {
+
+    public static final String CONTENT_TYPE = "fixed";
+
+    public static class Defaults extends NumberFieldMapper.Defaults {
+        public static final MappedFieldType FIELD_TYPE = new FixedPointFieldType();
+        public static final long DECIMAL_PLACES = 2;
+        public static final long DECIMAL_FACTOR = 100;
+
+        static {
+            FIELD_TYPE.freeze();
+        }
+    }
+
+    public static class Builder extends NumberFieldMapper.Builder<Builder, FixedPointFieldMapper> {
+
+        public Builder(String name) {
+            super(name, Defaults.FIELD_TYPE, Defaults.PRECISION_STEP_64_BIT);
+            builder = this;
+        }
+
+        public Builder nullValue(long nullValue) {
+            this.fieldType.setNullValue(nullValue);
+            return this;
+        }
+
+        public Builder decimalPlaces(long decimalPlaces) {
+            fieldType().setDecimalPlaces(decimalPlaces);
+            return this;
+        }
+
+        @Override
+        public FixedPointFieldType fieldType() {
+            return (FixedPointFieldType)fieldType;
+        }
+
+        @Override
+        public FixedPointFieldMapper build(BuilderContext context) {
+            setupFieldType(context);
+            FixedPointFieldMapper fieldMapper = new FixedPointFieldMapper(name, fieldType, defaultFieldType,
+                ignoreMalformed(context), coerce(context), context.indexSettings(), multiFieldsBuilder.build(this, context), copyTo);
+            return (FixedPointFieldMapper) fieldMapper.includeInAll(includeInAll);
```

Comment:
I agree this is weird in terms of API but was important to make mappings immutable